src/test/java/teammates/test/cases/logic/EmailsTest.java

```diff
@@ -436,6 +451,79 @@ private void verifyEmail(InstructorAttributes i, MimeMessage email,
     public void testNoExceptionThrownWhenNoMessagesToSend() {
         new Emails().sendEmails(new ArrayList<MimeMessage>());
     }
+    
+    @Test
+    public void testParseMimeMessageToSendgrid() throws MessagingException, JSONException, IOException {
+        if (Config.isUsingSendgrid()) {
```

Comment:
Same comment as above. In any case, reversing the condition can reduce nesting. e.g.
```
if(!Config.isUsingSendgrid()) { return; }
```